tools/windows-node-installer/test/e2e/azure_test.go

```diff
@@ -0,0 +1,226 @@
+}
+
+// isNil is a helper functions which checks if the object is a nil pointer or not.
+func isNil(v interface{}) bool {
+	return v == nil || (reflect.ValueOf(v).Kind() == reflect.Ptr &&
+		reflect.ValueOf(v).IsNil())
+}
+
+// setup initializes the azureProvider to be used for running the tests.
+func setup() (err error) {
+	var vmClientPtr *compute.VirtualMachinesClient
+	var nsgClientPtr *network.SecurityGroupsClient
+	oc, err := client.GetOpenShift(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("failed to initialize OpenShift client with error: %v", err)
+	}
+	provider, err := oc.GetCloudProvider()
+	if err != nil {
+		return fmt.Errorf("failed to get cloud provider information with error: %v", err)
+	}
+	resourceAuthorizer, err := auth.NewAuthorizerFromFileWithResource(azure.PublicCloud.ResourceManagerEndpoint)
+	if err != nil {
+		return fmt.Errorf("failed to get azure authorization token with error: %v", err)
+	}
+	getFileSettings, err := auth.GetSettingsFromFile()
+	if err != nil {
+		return fmt.Errorf("failed to get info from AZURE_AUTH_LOCATION with error: %v", err)
+	}
+	subscriptionId := getFileSettings.GetSubscriptionID()
+	if subscriptionId == "" {
+		return fmt.Errorf("failed to get the subscriptionId from the AZURE_AUTH_LOCATION")
+	}
+	vmClient := compute.NewVirtualMachinesClient(subscriptionId)
+	vmClientPtr = &vmClient
+	if vmClientPtr == nil {
+		return fmt.Errorf("failed to initialize the vm client")
+	}
+	vmClient.Authorizer = resourceAuthorizer
+	nsgClient := network.NewSecurityGroupsClient(subscriptionId)
+	nsgClientPtr = &nsgClient
+	if nsgClientPtr == nil {
+		return fmt.Errorf("failed to initialize the network security group client")
+	}
+	nsgClient.Authorizer = resourceAuthorizer
+	azureInfo.resourceGroupName = provider.Azure.ResourceGroupName
+	azureInfo.vmClient = vmClient
+	azureInfo.nsgClient = nsgClient
+	return nil
+}
+
+// readInstallerInfo reads the instanceIDs and secGroupsIDs from the
+// windows-node-installer.json file specified in "dir".
+func readInstallerInfo() (err error) {
+	wniFilePath := dir + "/windows-node-installer.json"
+	installerInfo, err := resource.ReadInstallerInfo(wniFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read installer info %s from ARTIFACT_DIR", dir)
+	}
+	if len(installerInfo.SecurityGroupIDs) == 0 {
+		return fmt.Errorf("failed to obtain the sec group Ids")
+	}
+	secGroupsIDs = installerInfo.SecurityGroupIDs
+	if len(installerInfo.InstanceIDs) == 0 {
+		return fmt.Errorf("failed to obtain the instance Ids")
+	}
+	instanceIDs = installerInfo.InstanceIDs
+	return nil
+}
+
+// isWinRMPortOpen returns a bool response if winRmHttps port is opened on
+// windows instance or not.
+func isWinRMPortOpen(secGroupRules []network.SecurityRule) bool {
+	var secRulePropFormat network.SecurityRulePropertiesFormat
+	var destPortRange string
+	var access network.SecurityRuleAccess
+	var protocol network.SecurityRuleProtocol
+	var priority int32
+	var sourceAddressPrefix string
```

Comment:
Why do need these variables defined in this manner? Can't they happen in the scope where they are being used?